Simplified/NYPLUserAccount.swift

```diff
@@ -89,7 +249,18 @@ extension Notification.Name {
   func hasCredentials() -> Bool {
     return hasAuthToken() || hasBarcodeAndPIN()
   }
-    
+
+  // Oauth requires login to load catalog
+  var isCatalogSecured: Bool {
+    return authDefinition?.isCatalogSecured ?? false
+  }
+
+  // MARK: - Legacy
+  
+  private var legacyBarcode: String? { return _barcode.read() }
+  private var legacyPin: String? { return _pin.read() }
+  var legacyAuthToken: String? { _authToken.read() }
```

Comment:
This is used for Oauth and SAML sessions to store auth token. We use this token to in requests to circulation manager

Reply:
so it's not really legacy and it's something we should keep in the long term?

Reply:
credentials enum took over the functionality of barcode, pin and auth token and extended it. Depending on the auth method, you can not have all of them. They are kept for convenience in case user still has some values stored in keychain with those old keys